### ISSUE TITLE
Improving "FA100 / future-rewritable-type-annotation" message

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#10573]
The message for rule https://docs.astral.sh/ruff/rules/future-rewritable-type-annotation/ `Missing from __future__ import annotations, but uses {name}` is slightly confusing / misleading. It sounds like I need to use `from __future__ import annotations` when using `typing.Union`, when it actually means that adding `from __future__ import annotations` *can* lead to simplified /modernized syntax. But that's also not checked unless relevant pyupgrades rules are turned on.

I think the message could be tweaked a bit to reflect that. I also assume FA and UP rules will likely be grouped together after the rules categorization #1774

**Comment by Olegt0rr** [comment]
The same issue.
`FA100` false positive detection

python 3.9

```python
from typing import Optional

from aiohttp import ClientSession

class Client:
    def __init__(self) -> None:
        self._session: Optional[ClientSession] = None
```

`Optional` is highlighted with `FA100 Missing `from __future__ import annotations`, but uses `typing.Optional``
